### ISSUE TITLE
Cannot parse JS line comment

### ISSUE THREAD

**Issue by Ayc0** [mdx-js/mdx#2079]
### Initial checklist

- [X] I read the [support docs](https://mdxjs.com/community/support/)
- [X] I read the [contributing guide](https://mdxjs.com/community/contribute/)
- [X] I agree to follow the [code of conduct](https://github.com/mdx-js/.github/blob/HEAD/code-of-conduct.md)
- [X] I searched [issues](https://github.com/search?q=user%3Amdx-js&type=Issues) and couldn’t find anything (or linked relevant results below)

### Affected packages and versions

MDX 2

### Steps to reproduce

If you open https://mdxjs.com/playground/ and try this code:

```mdx
<div
  {/* comment */}
>
  I am JSX
</div>
```

The compiler will crash with this error:

```
example.mdx
  2:4  error  Unexpected empty expression  unexpected-empty-expression  micromark-extension-mdx-expression

✖ 1 error
```

And if you try:

```mdx
<div
  // comment
>
  I am JSX
</div>
```

it'll crash with 

```
example.mdx
  2:4  error  Unexpected character `/` (U+002F) after self-closing slash, expected `>` to end the tag (note: JS comments in JSX tags are not supported in MDX)  unexpected-character  micromark-extension-mdx-jsx

✖ 1 error
```

Block | Line
-|-
<img width="1016" alt="image" src="https://user-images.githubusercontent.com/22725671/178146654-b5cc89e7-2c50-4f25-9891-ef8e4498b9d8.png"> | <img width="968" alt="image" src="https://user-images.githubusercontent.com/22725671/178146727-8ef4a76d-81bb-4f00-962c-5c7b7737009d.png">

### Expected behavior

In MDX 1, the block comment syntax wasn't supported, but line one was (it followed JSX's spec):

Line | Block
-|-
<img width="1440" alt="image" src="https://user-images.githubusercontent.com/22725671/178146761-71efa2ac-cb76-4b87-9200-0f6d55291ef8.png"> |<img width="1440" alt="image" src="https://user-images.githubusercontent.com/22725671/178146782-40ae7f5e-ab38-4f29-b25d-9134b908e97e.png">

Would it be possible add back this syntax in MDX 2?

### Actual behavior

It crashes for all form of comments

**Comment by wooorm** [comment]
Correct, those are not supported. It's a small deviation from jsx that not a lot of people know even exists, and not a lot of people use, that significantly makes parsing jsx harder.

**Comment by wooorm** [comment]
Closing as this is not planned, and rather complex. If you (or anyone stumbling on this) is interested in working on it, I can help advise, the work will have to be done in https://github.com/micromark/micromark-extension-mdx-jsx. You’d need to test a bit how different parsers (TS, Babel) handle comments in JSX, but essentially I believe it’s similar to whitespace, which is allowed almost everywhere. [Some examples in the Babel playground](https://babeljs.io/repl#?browsers=defaults%2C%20not%20ie%2011%2C%20not%20ie_mob%2011&build=&builtIns=false&corejs=3.21&spec=false&loose=false&code_lz=DwQw9AVBYEYAQGMC0loBM4FNVgLwCIAzfOAc1x3wAsSwA-IA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=env%2Creact%2Cstage-2&prettier=false&targets=&version=7.18.8&externalPlugins=&assumptions=%7B%7D).

Even more complex is that we retain comments in the tree, but we can’t do that for comments in JSX tags. Can that be solved? This is also a good reason not to support them: if they could be parsed, people would try to use `// eslint-ignore ...` comments in there, but we can likely not make that work.
